controllers/release/release_adapter.go

```diff
@@ -228,24 +229,36 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 // EnsureTargetContextIsSet is an operation that will ensure that the targetContext of the adapter is not nil. It will
 // set it to a context from the workspace in the ReleasePlan if one is defined or to its own context if not.
 func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
-	if a.targetContext == nil {
+	// If the target namespace/workspace is not set we should get the target from the ReleasePlan
+	if a.release.Status.Target == (kcp.NamespaceReference{}) {
+		patch := client.MergeFrom(a.release.DeepCopy())
 		releasePlan, err := a.getReleasePlan()
 		if err != nil {
-			patch := client.MergeFrom(a.release.DeepCopy())
 			a.release.MarkInvalid(v1alpha1.ReleaseReasonReleasePlanValidationError, err.Error())
 			return results.RequeueOnErrorOrStop(a.client.Status().Patch(a.context, a.release, patch))
 		}
-
-		if releasePlan.Spec.Target.Workspace == "" {
-			a.logger.Info("Running in a traditional kubernetes environment")
-			a.targetContext = a.context
-		} else {
+		if releasePlan.Spec.Target.Workspace != "" {
```

Comment:
You already know this is empty as `Target` is nil.

Reply:
My entry point checks for `a.release.Status.Target == (kcp.NamespaceReference{}) ` which could be false as well if `Workspace` is not set, but namespace is. I will change to check only for the Workspace to avoid redoing this check.

Reply:
Indeed. I refactored this a bit as I was checking this 3 times. I will resolve this comment so you can re-check this.